src/types/sequence.rs

```diff
@@ -287,12 +287,14 @@ macro_rules! array_impls {
                 fn extract(obj: &'source PyAny) -> PyResult<Self> {
                     let mut array = [T::default(); $N];
                     // first try buffer protocol
-                    if let Ok(buf) = crate::buffer::PyBuffer::get(obj) {
-                        if buf.dimensions() == 1 && buf.copy_to_slice(obj.py(), &mut array).is_ok() {
+                    if unsafe { ffi::PyObject_CheckBuffer(obj.as_ptr()) } == 1 {
+                        if let Ok(buf) = crate::buffer::PyBuffer::get(obj) {
```

Comment:
I guess that this ultimately results in using `PyObject_CheckBuffer` twice? I wonder if we can have a `pub(crate) fn get_unchecked` method for `PyBuffer` to avoid this.

Reply:
I don't think so? `PyBuffer::get` doesn't call `PyObject_CheckBuffer` internally.

Reply:
👍 ah ok